src/gui/XournalppCursor.cpp

```diff
@@ -307,10 +307,17 @@ void XournalppCursor::updateCursor() {
 
 
 auto XournalppCursor::getEraserCursor() -> GdkCursor* {
-    this->currentCursor = CRSR_ERASER;
 
     // Eraser's size follow a quadratic increment, so the cursor will do the same
     double cursorSize = control->getToolHandler()->getThickness() * 2 * control->getZoomControl()->getZoom();
```

Comment:
Please change this to `2.0`.

Reply:
@LittleHuba Could you please explain me, (if and) why that makes any difference? As I understand, the int 2 will be casted to a double implicitly. Should I always write integer values with a decimal point when it is multiplied with (or added to/subtracted from, divided by) a double?

Reply:
This is just a maintenance and readability thing. It's pretty annoying to have to chase down all of the type definitions just to figure out if/when there are casts. If we happen to change the type signature (which is possible, but unlikely), we can immediately avoid buggy implicit conversions.

Reply:
@Technius not necessarily...

What you will get is not as easy as "it will be cast to double". Depending on where the `int` is, the exact opposite can happen and you end up with casting all the operands to `int` and then multiply them to finally cast them back to `double` for storage. 

Additionally you might introduce overflows if you use `int` and `long` within the same formula. 

Generally stick with the biggest type (`int + long -> long`) and the most precise (`int + float -> float`) to not run in these issues. If you need to cast, do so after you did all the arithmetics and only need to store the result.

Reply:
Right, which is why we want to either make every cast explicit or eliminate casting altogether. Rounding errors are easily exposed when everything is explicit.